packages/plugin-cosmos/src/providers/wallet.ts

```diff
@@ -0,0 +1,96 @@
+import { SigningCosmWasmClient } from "@cosmjs/cosmwasm-stargate";
+import { DirectSecp256k1HdWallet } from "@cosmjs/proto-signing";
+import { GasPrice } from "@cosmjs/stargate";
+import { chains, assets } from "chain-registry";
+import type { IAgentRuntime, Provider, Memory, State } from "@ai16z/eliza";
+
+export class CosmosWalletProvider {
+    private wallet: DirectSecp256k1HdWallet;
+    private address: string;
+    private activeChain: string;
+
+    constructor(private mnemonic: string) {}
+
+    async initialize(chainName: string) {
+        const { prefix } = await this.getChainConfigs(chainName);
+        this.wallet = await DirectSecp256k1HdWallet.fromMnemonic(
+            this.mnemonic,
+            { prefix }
+        );
+        const [account] = await this.wallet.getAccounts();
+        this.address = account.address;
+        this.activeChain = chainName;
+    }
+
+    private async getChainConfigs(chainName: string) {
```

Comment:
why is this async?

Reply:
This function is based on `getChainSettings` from your's PR, which was also async :smile:. I'll change it.